From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Caner Altinbasak <cal@brightsign.biz>
Date: Tue, 29 Jul 2025 15:52:00 +0100
Subject: [PATCH] fix: Remove unsupported 'inputs' in GN tool() definitions

The Electron source tree uses 'inputs = rustc_wrapper_inputs' in
tool() blocks, which requires a newer GN version than what we build
from Chromium 147 sources. Remove these and the variable definition
to maintain compatibility.

Upstream-Status: Inappropriate [OE-specific]
---
 build/toolchain/gcc_toolchain.gni | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/build/toolchain/gcc_toolchain.gni b/build/toolchain/gcc_toolchain.gni
--- a/build/toolchain/gcc_toolchain.gni
+++ b/build/toolchain/gcc_toolchain.gni
@@ -677,13 +677,6 @@
       rustc = "$rust_compiler_prefix${rustc_bin}"
       rustc_wrapper += " --rustc=$rustc"
 
-      rustc_wrapper_inputs = [
-        "//build/action_helpers.py",
-        "//build/gn_helpers.py",
-        "//build/rust/gni_impl/rustc_wrapper.py",
-        "//third_party/rust-toolchain/bin/rustc",
-      ]
-
       # RSP manipulation due to https://bugs.chromium.org/p/gn/issues/detail?id=249
       tool("rust_staticlib") {
         libname = "{{output_dir}}/{{target_output_name}}{{output_extension}}"
@@ -708,7 +701,6 @@
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $libname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_rlib") {
@@ -741,7 +733,6 @@
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link,metadata=$rmetaname -o $rlibname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_bin") {
@@ -763,7 +754,6 @@
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $exename -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_cdylib") {
@@ -786,7 +776,6 @@
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $dllname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_macro") {
@@ -809,7 +798,6 @@
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $dllname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
     }
 
-- 
2.39.5

